PowerShell/BuildingBlocks/SDKOperations/RetrieveEntities.ps1

```diff
@@ -0,0 +1,39 @@
+# Initialize the context
+.\Init.ps1
+
+# Here are 3 common ways to retrieve entities
+# 1. Retrieve all the customers in a workgroup and process them one by one
+# Use -RetrieveAll and piping with ForEach 
+Get-BT_Customer -Ticket $mspc.WorkgroupTicket -RetrieveAll | ForEach {
+    # Process each customer retrieved
+    Write-Host $_.CompanyName
+}
+
+# 2. Retrieve and process customers from a workgroup in batches
+# Use paging
+$pageSize = 100
+$count = 0
+While( $true ) {   
+    # Retrieve customers in batch
+    [array]$batch = Get-BT_Customer -Ticket $mspc.WorkgroupTicket -PageOffset $($count * $pageSize)
+    if ( -not $batch ) { 
+        break
+    }
+    
+    # Update each customer in the batch
+    ForEach($customer in $batch) {
+        $customer.CountryName = 'USA'
+    }
+
+    # Send update request
+    Set-BT_Customer -Ticket $mspc.WorkgroupTicket -Customer $batch
```

Comment:
Same, that won't work, the ticket should be scoped to the customer, not the workgroup.
Since you are processing a batch of customers, you would need a different ticket for each customer (each of the ticket would be scoped to the corresponding customer).

Maybe we should use another entity for this sample script, to simplify it (e.g. Endpoint, which requires a customer-scoped ticket):
Set-BT_Endpoint -Ticket $mspc.CustomerTicket - Endpoint $batch

Reply:
OK